Sources/CvSelectionGroupAI.cpp

```diff
@@ -741,10 +725,13 @@ CvUnit* CvSelectionGroupAI::AI_getBestGroupAttacker(const CvPlot* pPlot, bool bP
 		// Recalculate, this time modifying the predicted hitpoints that will result
 		// if the battle occurs.  This will return a better odds estimate than we
 		//	had previously, but should not change the unit choice
-		iBestOdds = pBestUnit->AI_attackOdds(pPlot, bPotentialEnemy, ppDefender);
+		iUnitOdds = pBestUnit->AI_attackOdds(pPlot, bPotentialEnemy, ppDefender);
 	}
-	
-	iUnitOdds = iBestOdds;
+	else
+	{
+		iUnitOdds = iBestValue;
```

Comment:
My code is quite different for this function, did you change it already in your branch?